### PR TITLE
Adds authorization function that supports user whitelisting

### PR DESCRIPTION
**Pull request by dposada**

### CODE REVIEW

File: scheduler/src/cook/authorization.clj [hunk 1]
```diff
@@ -144,6 +145,47 @@
                      "is not allowed to perform" verb "on" (str object ",") "denying.")
            false))))
 
+(defn admins-open-gets-auth
```
Comment (wyegelwel) [inline scheduler/src/cook/authorization.clj]: Make it clear this also includes checking a whitelist

File: scheduler/src/cook/authorization.clj [hunk 2]
```diff
@@ -144,6 +145,47 @@
...
+  "Allows admins to do anything to any object, and if whitelisted-user is truthy,
+  allows non-admins to manipulate objects that they own and to read (get) any object."
+  [admins user verb {:keys [owner] :as object} whitelisted-user]
```
Comment (wyegelwel) [inline scheduler/src/cook/authorization.clj]: `whitelisted-user` implies it contains a user name. Rename it to something like `user-passed-whitelist?`

File: scheduler/src/cook/authorization.clj (hunk 1 above)
Comment (dposada) [inline scheduler/src/cook/authorization.clj]: Done

File: scheduler/src/cook/authorization.clj (hunk 2 above)
Comment (dposada) [inline scheduler/src/cook/authorization.clj]: Renamed to `user-in-whitelist?`

**Comment by dposada** [comment]
@wyegelwel Any other feedback on this?

**Pull request merged**